exporters/trace/src/main/java/com/google/cloud/opentelemetry/trace/NoopTraceExporter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2023 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.trace;
+
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.util.Collection;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A noop implementation of a {@link SpanExporter}. This class is implemented as a singleton and
+ * return a successful result on all public methods without performing any action.
+ */
+@ThreadSafe
+public class NoopTraceExporter implements SpanExporter {
```

Comment:
Our primary class is called `TraceExporter` because it exports to the `Cloud Trace` product. In this case, `NoopSpanExporter` is a more appropriate name.

Also, is there any reason to make this public rather than package-private?

Reply:
No, I will change it to package private, it was an oversight.